### PR TITLE
Httpx backend

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
Add [`httpx`](https://www.encode.io/httpx/) backend.

### CODE REVIEW

**Comment by brettcannon** [comment]
Looks like mypy isn't happy yet. :)

**Comment by yeraydiazdiaz** [comment]
Mypy deserves to be happy, even if the errors were in the tornado backend 😄 

Also had to fix the sphinx config removing looking for `docs/_static` to get CI to pass, another option is to add it even if it's empty, but I opted for removing it. Happy to change it though.

File: gidgethub/tornado.py [hunk 1]
```diff
@@ -11,11 +11,14 @@ class GitHubAPI(gh_abc.GitHubAPI):
...
         """Make an HTTP request."""
-        if method == "GET" and not body:
-            real_body = None
-        else:
-            real_body = body
-        request = httpclient.HTTPRequest(url, method, headers, real_body)
+        # passing body = None raises an error in mypy:
+        # Argument 4 to "HTTPRequest" has incompatible type "Optional[bytes]";
+        # expected "Union[bytes, str]"
+        request = (
```
Comment (brettcannon) [inline gidgethub/tornado.py]: Could you change this to the following?
```python
# Setting 'body' to None fails type checking, so only add a 'body' argument if necessary.
args = [url, method, dict(headers)]
if method == "GET" and not body:
    args.append(body)
request = httpclient.HTTPRequest(*args)
```
I think this is a bit easier to read and help communicate the situation a bit more.

Comment (yeraydiazdiaz) [reply gidgethub/tornado.py]: Sure, I've tried a couple of iterations of this but unfortunately mypy checks keep failing.

```
✗ mypy --ignore-missing-imports --strict gidgethub/*.py
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "str"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Union[Dict[str, str], HTTPHeaders]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Union[bytes, str]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "float"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Union[float, datetime]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "bool"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "int"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Callable[[bytes], None]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Callable[[str], None]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Callable[[Any], None]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Callable[[Callable[[bytes], None]], Future[None]]"
gidgethub/tornado.py:18: error: Argument 1 to "HTTPRequest" has incompatible type "*List[Union[str, Dict[Any, Any], bytes]]"; expected "Union[Dict[str, Any], SSLContext]"
Found 12 errors in 1 file (checked 8 source files)
```

It seems the unpacking of the args list is not sitting well with mypy. I have a feeling it expects the typing of `args` to match all possible combinations of arguments to [Tornado's HTTPRequest signature]https://www.tornadoweb.org/en/stable/_modules/tornado/httpclient.html#HTTPRequest) which include internal types and combinations of callables.

Seems like a lot of work for gidgethub.

Any ideas?

Comment (brettcannon) [reply gidgethub/tornado.py]: Yeah, just shut off mypy for that line. 😁 Try adding `# type: ignore` to the call line and see if that solves it (with a comment explaining why we're skipping the type check).

**Comment by brettcannon** [comment]
@yeraydiazdiaz and I forgot to mention that when you're ready for me to look at this again just re-request my review (circular arrow by my name under "Reviewers" in case you've never done it before).

**Comment by brettcannon** [comment]
@yeraydiazdiaz thanks! I've added doing a release to my TODO list (hopefully next Friday).

**Pull request merged**